### PR TITLE
Issue #27898

### PR DESCRIPTION
**Pull request by laoneo**
Added a check if the buffer from the document is really an array. Issue
number is #27898 on joomlacode.

### CODE REVIEW

**Comment by laoneo** [comment]
The cache class needs an array from $buffer1 = $document->getBuffer(); but when a Ajax request is done with a json response the $buffer1 is a string.

File: libraries/joomla/cache/cache.php [hunk 1]
```diff
@@ -549,6 +549,9 @@ public static function setWorkarounds($data, $options = array())
 
 		// Get the modules buffer before component execution.
 		$buffer1 = $document->getBuffer();
+		if(!is_array($buffer1)){
```
Comment (realityking) [inline libraries/joomla/cache/cache.php]: You need a space after the if and the { goes on a new line.

File: libraries/joomla/cache/cache.php [hunk 2]
```diff
@@ -605,6 +608,9 @@ public static function setWorkarounds($data, $options = array())
 			// @todo Check if the following is needed, seems like it should be in page cache
 			// Get the module buffer after component execution.
 			$buffer2 = $document->getBuffer();
+			if(!is_array($buffer2)){
```
Comment (realityking) [inline libraries/joomla/cache/cache.php]: Same as above.

**Comment by LouisLandry** [comment]
Thanks a bunch!

**Pull request merged**